src/components.test.js

```diff
@@ -0,0 +1,105 @@
+import renderer from 'react-test-renderer';
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Calculator from './components/Calculator';
+import App from './App';
+import Quote from './components/Quotes';
+import HomePage from './components/Home';
+
+const Heading = 'Math Magicians';
+const home = 'Math Magicians';
+const quote = 'Mathematics is not about numbers, equations, computations, or algorithems: it is about understanding. -William Paul Thurston';
+const calculatorHeading = 'Lets do some math!';
+
+describe('test rendering', () => {
+  test('Test for page rendering', () => {
+    const tree = renderer.create(<App />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+describe('navigation test', () => {
+  test('test if home is displayed', () => {
+    render(<App />);
+    const titles = screen.getAllByRole('heading');
+    expect(titles[0]).toHaveTextContent(Heading);
+    expect(titles[1]).toHaveTextContent(home);
+    expect(titles.indexOf(quote)).toBe(-1);
+    expect(titles.indexOf(calculatorHeading)).toBe(-1);
+  });
+
+  test('test if home is displayed', () => {
+    render(<HomePage />);
+    const titles = screen.getAllByRole('heading');
+    expect(titles[0]).toHaveTextContent(home);
+    expect(titles.indexOf(quote)).toBe(-1);
+    expect(titles.indexOf(calculatorHeading)).toBe(-1);
+  });
+
+  test('test if the quotes page works', () => {
+    render(<Quote />);
+    const titles = screen.getAllByRole('heading');
+    expect(titles[0]).toHaveTextContent(quote);
+    expect(titles.indexOf(home)).toBe(-1);
+    expect(titles.indexOf(calculatorHeading)).toBe(-1);
+  });
+
+  test('test naviation to calculator', () => {
+    render(<Calculator />);
+    const titles = screen.getAllByRole('heading');
+    expect(titles[0]).toHaveTextContent(calculatorHeading);
+    expect(titles.indexOf(quote)).toBe(-1);
+    expect(titles.indexOf(home)).toBe(-1);
+  });
+});
+
+describe('Test basic computations', () => {
```

Comment:
- [ ] As stated by the previous reviewer, and it's a project requirement, kindly solve these tests which are not passing.
![image](https://user-images.githubusercontent.com/91486791/220905457-8399c8ae-518c-41ad-9e6b-95bd3003c1fa.png)
![image](https://user-images.githubusercontent.com/91486791/220905626-5a44ed93-8d86-4b1b-a05e-a8d5fcbea5f4.png)
* The reason why those tests are not passing is they can't stimulate click event as there is more than one of that specific element you have called. You can try calling it differently to solve this issue:
![image](https://user-images.githubusercontent.com/91486791/220906178-808fd5ce-e5f7-4f4a-927a-9e682239deda.png)
![image](https://user-images.githubusercontent.com/91486791/220906554-917b5416-ffa8-4568-bcb8-4a9d538461c2.png)
* 
![image](https://user-images.githubusercontent.com/91486791/220908238-f326e180-0abd-40bd-91c4-1b69b2d5b2e2.png)
I think the reason behind this failing one is you assign the wrong value to your result variable in the test case. The text you should choose is in the output bar and it is not set to 0 at the start. So it chooses the button 0 and it returns 0 in the end.
![image](https://user-images.githubusercontent.com/91486791/220908453-15749797-a108-4c51-bf36-424c2c6a4762.png)
![image](https://user-images.githubusercontent.com/91486791/220908988-76013d48-8bd4-4e03-9698-97fbd8bd819d.png)
![image](https://user-images.githubusercontent.com/91486791/220909254-85cdb52c-5e64-471a-8855-51ebd5b7f2ea.png)

* The last one is : 
![image](https://user-images.githubusercontent.com/91486791/220909648-ac9cff85-0aa0-47a5-b23e-a068ab7c24bd.png)

* I believe you can solve these issues. If you think this review is unfair you can ask for a second opinion. [this form](https://airtable.com/shrQAqnBwek5a0O0s)